Compiler/BackEnd/Differentiate.mo

```diff
@@ -2655,15 +2655,15 @@ algorithm
     String matrixName;
 
     case ({}, _, _, _, _, _, _, _)
-    then (inElementsDer, inFunctionTree, inElementsNoDer, inBooleanLst);
+    then (MetaModelica.Dangerous.listReverseInPlace(inElementsDer), inFunctionTree, MetaModelica.Dangerous.listReverseInPlace(inElementsNoDer), MetaModelica.Dangerous.listReverseInPlace(inBooleanLst));
```

Comment:
listReverseInPlace does not belong here: the function can be called with a non-empty input which is messed up.